### DISCUSSION TITLE
Write 2D wave spectra

### DISCUSSION THREAD

**Discussion by Hendrik1987** [DHI/mikeio#580] [Q&A] [answered]
Hi,

I think, I can define a mikeio dataarray holding a 2D wave spectrum. Does that look OK? I am wondering about the double up of "frequency" and "direction" in dimensions and geometry...

![image](https://github.com/DHI/mikeio/assets/29595985/70fcc316-4e59-4302-82d5-ec4a8efb2aba)

![image](https://github.com/DHI/mikeio/assets/29595985/608cd81a-71f6-41f3-b8ed-17757349b037)

1. How would I combine multiple point spectra to a line (I can't figure out how to initiate a `GeometryFMLineSpectrum`)
2. It appears that writing of spectra type geometries is not yet implemented https://github.com/DHI/mikeio/blob/66beb8d97a627bf848c72a359dac24e10a7af5b0/mikeio/dataset/_dataset.py#L1849
Or is there another way?

Thanks!

**Comment by Hendrik1987** [comment]
@jsmariegaard , @daniel-caichac-DHI, anyone any hints :yum:? Have a nice summer!

**Reply by jsmariegaard** [reply, marked as answer]
Hey @Hendrik1987, We still miss the basic writing functionality of spectral dfsu files in MIKE Core. That is the main stopping block. I did a project with a client a year ago where we needed the same for an operational system. In that case, we made a template file using MIKE 21 SW at the relevant positions and then replaced the data (and timesteps) using the GenericOpen functionality in MIKE Core. Would that work for you?

**Reply by Hendrik1987** [reply]
@jsmariegaard, yes I guess that would work! In the meantime we went .nc -> .dfs2 and then .dfs2 to .dfsu using Mike zero. But we had to interpolate to equidistant spectra first and the non 0 origin gave us some trouble. I think it’s related to #574. In short, I shall try your approach next time! :-)

**Reply by Hendrik1987** [reply]
@jsmariegaard : one follow up. Did you then match also the frequency and direction vectors in your template .dfsu file from the MIKE 21 SW run to match the source spectra you wanted to write? Or did you do some interpolation to spectra resolutions OK for MIKE 21 SW (if there are any limitations...)? Many thanks!

**Reply by jsmariegaard** [reply]
I changed the template MIKE21SW simulation spectral resolution to match the input. In the final run the engine will take care of the interpolation.
